cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/EOGWorklist.kt

```diff
@@ -248,9 +248,18 @@ inline fun <reified K : Node, V> iterateEOG(
     while (worklist.isNotEmpty()) {
         val (nextNode, state) = worklist.pop()
 
-        val newState = transformation(nextNode, state.duplicate(), worklist)
+        val insideBB =
+            (nextNode.nextEOG.size == 1 &&
+                nextNode.prevEOG.size == 1 &&
+                nextNode.prevEOG.first().nextEOG.size == 1)
```

Comment:
you could use `single` or `singleOrNull` instead of checking for size 1 and using first()